plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/wc-admin-helper.php

```diff
@@ -240,14 +244,18 @@ public function test_is_store_page() {
 
 		$wp_rewrite->expects( $this->any() )
 			->method( 'generate_rewrite_rule' )
-			->willReturn( array( 'shop/(.+?)/?$' => 'index.php?product_cat=$matches[1]&year=$matches[2]' ) );
+			->willReturn(
+				array(
+					'shop/.*?$' => 'index.php?product_cat=$matches[1]&year=$matches[2]',
```

Comment:
I'm curious why was this change needed?

Reply:
@ilyasfoo The previous regex did not catch "https://example.com/shop/?tag=test".

Reply:
Got it, thanks. However, I believe the regex doesn't seem to capture any group, thus `$matches[1]` and `$matches[2]` will never be defined. I'm wondering if this will have any unexpected outcome 🤔 

cc @chihsuan since I think you originally introduced the regex

Reply:
@ilyasfoo Good point. The tests are passing, so I also wonder if we still need them. I'll wait for Chi's input.

Reply:
Hey @moon0326 @ilyasfoo, The regex was introduced to catch the URL path when the Product permalinks is set to `/shop/%product_cat%/` or something like that.

For example, if the product permalink is `/shop/%product_cat%/`, then the URL path `/shop/test` will match the regex.

<img width="990" alt="Screenshot 2024-11-26 at 15 47 14" src="https://github.com/user-attachments/assets/5220619a-dbff-4a4c-b128-a4f4a4c2bbb8">

Reply:
Thanks for the input, @chihsuan!

```suggestion
					'shop/.*?$' => 'index.php',
```

@moon0326  It seems we might not need `$matches` after all. What do you think of simplifying it so it doesn't add confusion in the future? My local test is passing with this change.

Reply:
@ilyasfoo Sounds good!